harness/tests/common/api/test_session.py

```diff
@@ -0,0 +1,54 @@
+from determined.common import api
+from determined.common.api import certs
+from tests.common import api_server
+
+
+def test_session_non_persistent_http_sessions() -> None:
+    with api_server.run_api_server(address=("localhost", 8080)) as master_url:
```

Comment:
FWIW, I think `api_server.run_api_server` was a design mistake, and that we should either mock python endpoints or HTTP calls with `responses`. I think this would be plenty easy to do here, and would be clearer to the reader, in addition.

I don't think you _have_ to migrate, but I do think it'd be better for our test health to stop using the fake master.

Reply:
believe me i really didn't want to use this. tried both mocking endpoints and `responses`. the underlying HTTP connection pools only exist for real connections